File: C_Cursor.cs
using SFML.Graphics;
using SFML.System;

namespace Game_of_life
{
	class C_Cursor : Drawable
	{
		RectangleShape c_c;

		public Color color
		{
			get => c_c.OutlineColor;
			set => c_c.OutlineColor = value;
		}

		public void Draw(RenderTarget target, RenderStates states)
		{
			target.Draw(c_c, states);
		}


		public C_Cursor(float size, float grosor)
		{
			c_c = new RectangleShape(new Vector2f(size, size));
			c_c.OutlineThickness = grosor;
			c_c.FillColor = Color.Transparent;
			c_c.OutlineColor = Color.Red;
		}
		public void setPos(int x, int y)
		{
			int new_x = x/(int)c_c.Size.X;
			int new_y = y/(int)c_c.Size.X;
			c_c.Position = new Vector2f(new_x*c_c.Size.X, new_y*c_c.Size.X);
		}
	}
}

File: src/c_cursor.cpp
#include <c_cursor.hpp>

C_Cursor::~C_Cursor() { delete m_cursor; m_cursor = nullptr; }

C_Cursor::C_Cursor(float c_size, float o_size)
{
	m_cursor = new sf::RectangleShape({c_size, c_size});
	m_cursor->setFillColor(sf::Color::Transparent);
	m_cursor->setOutlineThickness(o_size);
	m_cursor->setOutlineColor(sf::Color::Red);
}

void C_Cursor::setPos(int x, int y)
{
	m_cursor->setPosition(x, y);
}

void C_Cursor::tick()
{
	if (m_cursor->getOutlineColor() == sf::Color::Red)
		m_cursor->setOutlineColor(sf::Color::Yellow);
	else
		m_cursor->setOutlineColor(sf::Color::Red);
}

void C_Cursor::draw(sf::RenderTarget &target, sf::RenderStates states) const
{
	target.draw(*m_cursor, states);
}
